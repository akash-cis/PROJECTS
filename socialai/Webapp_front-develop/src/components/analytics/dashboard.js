import styled from "styled-components"
import { Colors } from "../../library/constants"

export const TabsContainer = styled.div`
  display: flex;
  justify-content: center;
`

export const InlineTabs = styled.div`
  border-bottom: ${props =>
    props.active
      ? `2px solid ${Colors.primaryBrandBlue}`
      : "2px solid transparent"};
  margin: 0;
  cursor: pointer;
  line-height: 60px;
  margin-right: 1rem;
`

export const squashByKeySumByOther = (data, key, sumKey, cb = kpi => kpi) => {
  let newArray = []
  // we apply the callback just before pushing it
  const pushKpi = key => {
    newArray.push(cb(key))
  }
  data.forEach(kpi => {
    const cbKpi = cb(kpi)
    const sameNameKpi = newArray.find(newKpi =>
      newKpi && newKpi[key] ? newKpi[key] === cbKpi[key] : null
    )
    if (sameNameKpi) {
      let ob
      newArray = newArray.filter(newKpi => newKpi[key] !== sameNameKpi[key])
      if (cbKpi[sumKey] >= 0 && typeof sameNameKpi[sumKey] === "number") {
        if (sameNameKpi.hasOwnProperty("children")) {
          ob = {
            ...sameNameKpi,
            [sumKey]: sameNameKpi[sumKey] + cbKpi[sumKey],
            children: [
              ...sameNameKpi.children,
              { name: cbKpi.model, count: cbKpi.total },
            ],
          }
        } else {
          ob = {
            ...sameNameKpi,
            [sumKey]: sameNameKpi[sumKey] + cbKpi[sumKey],
            children: [{ name: cbKpi.model, count: cbKpi.total }],
          }
        }
      }
      if (ob) return pushKpi(ob)
    }
    pushKpi(cbKpi)
  })
  return newArray
}

const NAME = "name"
const TOTAL = "total"
const CHILDREN = "children"

export const toTreeMapCompatible = data => {
  if (!data) return null
  const newData = data.map(({ make, count, model }) => ({
    name: make,
    total: count,
    model: model,
  }))
  const squashed = squashByKeySumByOther(newData, NAME, TOTAL, item => {
    if (!item.hasOwnProperty(CHILDREN)) {
      return { ...item, children: [{ name: item.model, count: item.total }] }
    }
    return item
  })
  return squashed
}
