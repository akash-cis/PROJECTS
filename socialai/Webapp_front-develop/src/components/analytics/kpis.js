import React from "react"
import styled from "styled-components"
import { Menu } from "antd"
import Typography from "../../library/typography"
import { UnstatedTabNav } from "../../library/tabs"
import { TabsContainer } from "../../library/basicComponents"
import Up from "../../icons/svg/Caret/Filled/Down-Red.svg"
import Down from "../../icons/svg/Caret/Filled/Up-Green.svg"
import UserRoleProtected from "../userRoleProtected"

const toSlug = item =>
  item
    .toLowerCase()
    .split(" ")
    .join("-")
const NavItems = [
  { title: "KPIs", slug: "kpis" },
  { title: "App Usage", slug: "usage" },
  {
    title: "General Dashboard",
    slug: "dashboard",
    role: "canViewAutoAnalytics",
  },
]

export const AnalyticsLayout = ({ location, children }) => {
  return (
    <>
      <TabsContainer>
        {NavItems.map(item =>
          item.role ? (
            <UserRoleProtected
              component={UnstatedTabNav}
              to={`/analytics/${toSlug(item.slug)}`}
              active={`/analytics/${toSlug(item.slug)}` === location.pathname}
              key={item.slug}
              userRole={item.role}
            >
              {item.title}
            </UserRoleProtected>
          ) : (
            <UnstatedTabNav
              to={`/analytics/${toSlug(item.slug)}`}
              active={`/analytics/${toSlug(item.slug)}` === location.pathname}
              key={item.slug}
            >
              {item.title}
            </UnstatedTabNav>
          )
        )}
      </TabsContainer>
      {children}
    </>
  )
}

export const Card = styled.div`
  & * {
    margin-bottom: 0.5rem;
  }
`

export const CardsContainer = styled.div`
  padding: 1.5rem;
  width: ${props => (props.width ? props.width : "auto")};
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  margin-bottom: 0;
  text-align: center;
  @media only screen and (max-width: 1024px) {
    justify-content: center;
  }
`

export const CardTitle = styled(props => (
  <Typography variant="h4" weight="medium" {...props} />
))`
  text-transform: uppercase;
  color: #5e5e5e;
`

export const CardChange = styled.span`
  border-radius: 5px;
  padding: 8px;
  background-color: ${props => (props.positive ? "#E2F9F0" : "#FFE6E6")};
  color: ${props => (props.positive ? "#3ED57F" : "#FF4852")};
  display: inline-block;

  &:last-child {
    margin-bottom: 0;
  }
`

export const createMenu = items => (
  <Menu>
    {items.map(item => (
      <Menu.Item>{item}</Menu.Item>
    ))}
  </Menu>
)

const Caret = styled.img`
  margin: 0;
  margin-right: 5px;
`

export const ProspectAnalytics = ({ data, width, compareType = false }) => (
  <CardsContainer width={width}>
    {data.map(
      ({
        title,
        count,
        change,
        previous,
        isPercentage = false,
        compareType,
      }) => {
        const isPositive = change > 0 || change === "*"

        if (compareType) {
          return (
            <Card key={title}>
              <CardTitle>{title}</CardTitle>
              <Typography weight="bold" variant="h1">
                {count} / {previous}
              </Typography>
            </Card>
          )
        } else {
          return (
            <Card key={title}>
              <CardTitle>{title}</CardTitle>
              <Typography weight="bold" variant="h1">
                {count}
                {isPercentage && "%"}
              </Typography>
              <CardChange positive={isPositive}>
                <Caret src={isPositive ? Up : Down} alt="change" />
                {change}% (vs. {previous})
              </CardChange>
            </Card>
          )
        }
      }
    )}
  </CardsContainer>
)
