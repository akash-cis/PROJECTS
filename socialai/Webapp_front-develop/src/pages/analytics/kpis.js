import React, { useContext } from "react"
import { Alert, Row } from "antd"
import { useQuery } from "@apollo/react-hooks"
import {
  AnalyticsLayout,
  ProspectAnalytics,
} from "../../components/analytics/kpis"
import PieChart from "../../library/pieChart"
import LineChart from "../../library/lineChart"
import BarChart from "../../library/barChart"
import GaugeChart from "../../library/gaugeChart"
import OptionsWrapper from "../../library/optionsWrapper"
import {
  GET_ANALYTICS_KPIS,
  GET_APP_USAGE,
  GET_APP_USAGE_ALL,
} from "../../graphql/query"
import {
  prospectAnalysis,
  countByStatus,
  teamManagement,
  prospectUsageOverTime,
  prospectUsage,
  getTotal,
  teamLeaderboard,
  engagedConverted,
  ENGAGED_SPEC,
} from "../../components/analytics/lambdas"
import { UserContext } from "../../amplify/authenticator-provider"
import TeamPicker from "../../library/teamPicker"
import TeamLeaderboard from "../../library/teamLeaderboard"
import { BAR, LINE } from "../../library/constants"
import { ContainerGroup } from "../../library/basicComponents"
// Mocked data
// const cardsSold = [
//   { date: "Used Cards", value: 70 },
//   { date: "New Cards", value: 30 },
// ]

// const leadsData = [
//   {
//     title: "Leads Provided",
//     amount: "18,283",
//     change: "18",
//     previous: "12,195",
//   },
//   {
//     title: "Engaged With",
//     amount: "10,000",
//     change: "-18",
//     previous: "12,195",
//   },
//   {
//     title: "Leads converted",
//     amount: "7,000",
//     change: "18",
//     previous: "12,195",
//   },
// ]

// This section is pretty weird sorry guys!
// Most of the logic can be better if done on the backend side instead of the frontend

// OptionsWrapper is the main component here. Please search for it if debugging is needed.
const AnalyticsPage = ({ location }) => {
  const [data, setData] = React.useState()
  const { user } = useContext(UserContext)
  const isTeamMember = user && user.teamsLeader && user.teamsLeader.length === 0
  const isTeamLeader = user && user.teamsLeader && user.teamsLeader.length > 0

  const { data: analyticsData, loading, error } = useQuery(GET_ANALYTICS_KPIS, {
    variables: {
      range: 180,
      teamId: isTeamLeader ? user.teamsLeader[0]["id"] : -1,
    },
    fetchPolicy: "network-only",
  })

  React.useEffect(() => {
    if (!loading && !error) {
      setData({
        analytics: analyticsData.getAnalyticsKpis,
      })
      if (isTeamLeader && analyticsData.team && analyticsData.team.members) {
        const team = analyticsData.team.members.map(({ member }) => ({
          value: member.id,
          text: member.fullName,
        }))
        setData(data => ({
          ...data,
          team,
        }))
      }
    }
  }, [loading, error])

  return (
    <AnalyticsLayout location={location}>
      {!error && (
        <>
          {isTeamMember && (
            <IndividualView loading={loading} data={data && data.analytics} />
          )}
          {isTeamLeader && (
            <ManagerView
              data={data && data.analytics}
              team={data && data.team}
              loading={loading}
            />
          )}
        </>
      )}
      {error && (
        <Alert
          message={"Something went wrong. Please contact support."}
          type={"error"}
          banner
        />
      )}
    </AnalyticsLayout>
  )
}

const IndividualView = ({ data, loading }) => {
  const {
    data: analyticsData,
    loading: teamLoading,
    error: teamError,
  } = useQuery(GET_ANALYTICS_KPIS, {
    variables: {
      range: 180,
      teamId: -1,
      allMembers: true,
    },
    fetchPolicy: "network-only",
  })

  return (
    <>
      {/*Prospect Analytics*/}
      <Row>
        <OptionsWrapper
          changes
          lambda={prospectAnalysis}
          data={data}
          loading={loading}
          title="Prospect analysis"
        >
          {({ data }) => <ProspectAnalytics data={data} width={"70%"} />}
        </OptionsWrapper>
      </Row>
      {/*Leads*/}
      <ContainerGroup
      // style={{ flexFlow: "row wrap", justifyContent: "space-between" }}
      >
        <OptionsWrapper
          span={8}
          periods
          data={data}
          loading={loading}
          lambda={data => ({
            value: countByStatus(data)(ENGAGED_SPEC),
            total: getTotal(data),
          })}
          title="Leads engaged with"
        >
          {({ data }) => (
            <GaugeChart
              action="Engaged with"
              type="provided leads"
              {...data}
              noBubble
              hand
            />
          )}
        </OptionsWrapper>
        <OptionsWrapper
          span={8}
          data={data}
          loading={loading}
          periods
          lambda={engagedConverted}
          title="Engaged/Converted"
        >
          {({ data }) => <PieChart id="-cards-sold" data={data} />}
        </OptionsWrapper>
        <OptionsWrapper
          span={8}
          loading={teamLoading}
          data={analyticsData?.getAnalyticsKpis}
          lambda={teamManagement}
          error={teamError}
          periods
          title="Team engagement"
        >
          {({ data }) => (
            <PieChart
              id="-team-management"
              middle
              info
              action="Total Engaged Leads"
              {...data}
            />
          )}
        </OptionsWrapper>
      </ContainerGroup>

      {/*Prospect usage over time*/}
      <OptionsWrapper
        periods
        lambda={prospectUsageOverTime}
        data={data}
        loading={loading}
        graphTypes={[LINE, BAR]}
      >
        {({ data, period, graphType }) => (
          <>
            {graphType === BAR && (
              <BarChart period={period} id="prospect" data={data} />
            )}
            {graphType === LINE && <LineChart data={data} />}
          </>
        )}
      </OptionsWrapper>
    </>
  )
}

const ManagerView = ({ data, loading, team }) => {
  const { data: teamData, loading: teamLoading, error: teamError } = useQuery(
    GET_APP_USAGE_ALL,
    {
      variables: {
        range: 180,
        teamId: -1,
        allMembers: true,
      },
      fetchPolicy: "network-only",
    }
  )

  return (
    <>
      {/*Prospect Analytics*/}
      <Row>
        <OptionsWrapper
          changes
          lambda={prospectAnalysis}
          data={data}
          loading={loading}
          title="Prospect analysis"
          team={team}
        >
          {({ data }) => <ProspectAnalytics data={data} width={"70%"} />}
        </OptionsWrapper>
      </Row>
      {/*Leads*/}
      <ContainerGroup>
        <OptionsWrapper
          span={8}
          periods
          data={data}
          loading={loading}
          lambda={data => ({
            value: countByStatus(data)(ENGAGED_SPEC),
            total: getTotal(data),
          })}
          title="Leads engaged with"
          showTeam={false}
          team={team}
        >
          {({ data, setSelectedTeam }) => (
            <GaugeChart action="Engaged with" type="provided leads" {...data}>
              <TeamPicker team={team} setOuter={setSelectedTeam} />
            </GaugeChart>
          )}
        </OptionsWrapper>
        <OptionsWrapper
          team={team}
          data={data}
          loading={loading}
          lambda={prospectUsage}
          span={16}
          periods
          title="Leads engaged with"
        >
          {({ data }) => <BarChart horizontal={true} data={data} />}
        </OptionsWrapper>
      </ContainerGroup>

      {/* Team Leaderboard */}
      <OptionsWrapper
        padded
        title="Team Leaderboard"
        data={teamData?.getAnalyticsAppUsage}
        loading={teamLoading}
        lambda={teamLeaderboard}
        periods
        scroll={true}
      >
        {({ data }) => <TeamLeaderboard data={data} />}
      </OptionsWrapper>

      {/*Leads*/}
      {/* <ContainerGroup>
      <OptionsWrapper span={8} periods title="Leads engaged (by type)">
        {props => (
          <PieChart id="-team-management" data={teamManagement} {...props} />
        )}
      </OptionsWrapper>
      <OptionsWrapper span={8} periods title="Leads engaged with">
        {props => (
          <PieChart id="-team-management2" data={teamManagement} {...props} />
        )}
      </OptionsWrapper>
      <OptionsWrapper span={8} periods title="Leads engaged/converted">
        {props => (
          <PieChart id="-team-management3" data={teamManagement} {...props} />
        )}
      </OptionsWrapper>
        </ContainerGroup*/}

      {/*Prospect usage over time*/}
      <OptionsWrapper
        periods
        team={team}
        lambda={prospectUsageOverTime}
        loading={loading}
        data={data}
        graphTypes={[LINE, BAR]}
      >
        {({ data, period, graphType }) => (
          <>
            {graphType === BAR && (
              <BarChart period={period} id="prospect" data={data} />
            )}
            {graphType === LINE && <LineChart data={data} />}
          </>
        )}
      </OptionsWrapper>
    </>
  )
}

export default AnalyticsPage
