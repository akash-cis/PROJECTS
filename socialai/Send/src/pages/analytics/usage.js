import React, { useContext } from "react"
import { Button } from "antd"
import LeveledBarChart, {
  BarChartProvider,
  BarChartContext,
} from "../../library/leveledBarChart"
import { Colors } from "../../library/constants"
import OptionsWrapper from "../../library/optionsWrapper"
import { AnalyticsLayout } from "../../components/analytics/kpis"
import { UserContext } from "../../amplify/authenticator-provider"
import LayeredChart from "../../library/layeredBarChart"
import { useQuery } from "@apollo/react-hooks"
import { GET_APP_USAGE_ALL, GET_APP_USAGE } from "../../graphql/query"
import { managerAppUsage } from "../../components/analytics/lambdas"
import Typography from "../../library/typography"
import {
  Container,
  ContainerNavigation,
  PaddedCol,
  Content,
  CenteredContainer,
  LoadingIcon,
  Flex,
} from "../../library/basicComponents"
import { useState } from "react"
import BarChartContextAwareSlider from "../../library/slider/BarChartContextAwareSlider"

// Mocked data
const data = [
  { date: "2020-01-31T00:00:00", time: 1200 },
  { date: "2020-01-23T00:00:00", time: 2300 },
  { date: "2020-02-06T00:00:00", time: 700 },
  { date: "2020-01-29T00:00:00", time: 6400 },
]

const layeredData = [
  {
    country: "USA",
    year2004: 4.5,
    year2005: 4.2,
  },
  {
    country: "UK",
    year2004: 1.7,
    year2005: 3.1,
  },
  {
    country: "Canada",
    year2004: 2.8,
    year2005: 2.9,
  },
  {
    country: "Japan",
    year2004: 2.6,
    year2005: 2.3,
  },
  {
    country: "France",
    year2004: 1.4,
    year2005: 2.1,
  },
  {
    country: "Brazil",
    year2004: 4.6,
    year2005: 4.9,
  },
]

const maxObj = (data, attr) =>
  data.reduce(
    (prev, current) => (prev[attr] > current[attr] ? prev : current),
    -Infinity
  )

const UsagePage = ({ location }) => {
  const { user } = useContext(UserContext)
  const isTeamMember = user && user.teamsLeader && user.teamsLeader.length === 0
  const isTeamLeader = user && user.teamsLeader && user.teamsLeader.length > 0
  const { data, loading } = useQuery(GET_APP_USAGE, {
    fetchPolicy: "network-only",
  })

  return (
    <BarChartProvider>
      <AnalyticsLayout location={location}>
        {isTeamMember && (
          <IndividualView data={data?.getAnalyticsAppUsage} loading={loading} />
        )}
        {isTeamLeader && (
          <ManagerView data={data?.getAnalyticsAppUsage} loading={loading} />
        )}
      </AnalyticsLayout>
    </BarChartProvider>
  )
}

const IndividualView = ({ data: appUsage, loading }) => {
  const [period, setPeriod] = useState("day")
  const context = useContext(BarChartContext)

  return (
    <PaddedCol>
      <Container auto noMargin={true}>
        <ContainerNavigation>
          <Typography weight="medium">App usage over time</Typography>
          <Flex>
            <BarChartContextAwareSlider context={context} />
            <Button.Group>
              <Button onClick={() => setPeriod("month")}>Monthly</Button>
              <Button onClick={() => setPeriod("week")}>Weekly</Button>
              <Button onClick={() => setPeriod("day")}>Daily</Button>
            </Button.Group>
          </Flex>
        </ContainerNavigation>
        <Content noFlex>
          {loading && (
            <CenteredContainer>
              <LoadingIcon type="loading" />
            </CenteredContainer>
          )}
          {appUsage && (
            <LeveledBarChart
              id={"barMock"}
              data={appUsage}
              period={period}
              level={{
                value: maxObj(appUsage, "time").time,
                color: Colors.orange,
                text: "Best",
              }}
              duration
            />
          )}
        </Content>
      </Container>
    </PaddedCol>
  )
}

const ManagerView = ({ data, loading }) => (
  <OptionsWrapper
    changes
    data={data}
    lambda={managerAppUsage}
    loading={loading}
    title="Team app usage"
  >
    {({ data: orderedData }) => {
      return <LayeredChart data={orderedData} />
    }}
  </OptionsWrapper>
)

export default UsagePage
