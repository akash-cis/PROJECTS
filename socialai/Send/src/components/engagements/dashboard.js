import React, { useState } from "react"
import { useQuery } from "@apollo/react-hooks"
import { Row, Col, Layout } from "antd"
import OptionsWrapper from "../../library/optionsWrapper"
import { ProspectAnalytics } from "../analytics/kpis"
import { ContainerGroup } from "../../library/basicComponents"
import {
  getChange,
  leadBySource,
  leadAnalysis,
  filterByStatus,
  responseByEngagement,
  activityAnalysis,
  appointmentsBySalesPerson,
} from "../analytics/lambdas"
import BarChart from "../../library/barChart"
import LineChart from "../../library/lineChart"
import { BAR, LINE } from "../../library/constants"
import {
  GET_ENGAGEMENT_ANALYTICS,
  GET_ENGAGEMENT_LEADS_ANALYTICS,
  GET_LEADS_ANALYTICS,
  GET_APPOINTMENT_ANALYSIS,
  GET_APPOINTMENTS_BY_SOURCE,
  GET_APPOINTMENTS_BY_SALES_PERSON,
} from "../../graphql/query"
import moment from "moment"


const currentDate = moment()
  .utc()
  .format()
const data = [
  {
    date: currentDate,
    userId: 168,
    status: "Accepted",
    count: 1,
    source: "Autotrader",
    userName: "James Anderson",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Viewed",
    count: 300,
    source: "Autotrader",
    userName: "Ameny Lee",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Viewed",
    count: 193,
    source: "Reddit",
    userName: "Jecica",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Viewed",
    count: 55,
    source: "Cars.com",
    userName: "James Anderson",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Deal Won",
    count: 502,
    source: "Cars.com",
    userName: "Ameny Lee",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 580,
    status: "Pushed to CRM",
    count: 409,
    source: "Reddit",
    userName: "Ameny Lee",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Viewed",
    count: 167,
    source: "Reddit",
    userName: "James Anderson",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Viewed",
    count: 290,
    source: "Cars.com",
    userName: "Justine Smith",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Viewed",
    count: 128,
    source: "Autotrader",
    userName: "Justine Smith",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Available",
    count: 2915,
    source: "Cars.com",
    userName: "ALL TEAM",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Available",
    count: 100,
    source: "Cars.com",
    userName: "ALL TEAM",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Pushed to CRM",
    count: 520,
    source: "Autotrader",
    userName: "Justine Smith",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Available",
    count: 100,
    source: "Autotrader",
    userName: "Justine Smith",
    __typename: "KPIResponse",
  },
  {
    date: currentDate,
    userId: 168,
    status: "Available",
    count: 250,
    source: "Autotrader",
    userName: "Ameny Lee",
    __typename: "KPIResponse",
  },
]

const responseData = [
  {
    date: currentDate,
    status: "Sent",
    count: 150,
    source: "Autotrader",
    attampt: "1st Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Sent",
    count: 409,
    source: "Autotrader",
    attampt: "2nd Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Sent",
    count: 405,
    source: "Autotrader",
    attampt: "3rd Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Sent",
    count: 300,
    source: "Reddit",
    attampt: "1st Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Sent",
    count: 300,
    source: "Reddit",
    attampt: "2st Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Sent",
    count: 300,
    source: "Reddit",
    attampt: "3rd Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Sent",
    count: 300,
    source: "Cars.com",
    attampt: "1st Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Sent",
    count: 300,
    source: "Cars.com",
    attampt: "2nd Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Sent",
    count: 300,
    source: "Cars.com",
    attampt: "3rd Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Received",
    count: 193,
    source: "Autotrader",
    attampt: "1st Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Received",
    count: 155,
    source: "Autotrader",
    attampt: "2nd Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Received",
    count: 105,
    source: "Autotrader",
    attampt: "3rd Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Received",
    count: 193,
    source: "Reddit",
    attampt: "1st Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Received",
    count: 155,
    source: "Reddit",
    attampt: "2nd Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Received",
    count: 105,
    source: "Reddit",
    attampt: "3rd Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Received",
    count: 193,
    source: "Cars.com",
    attampt: "1st Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Received",
    count: 155,
    source: "Cars.com",
    attampt: "2nd Attampt",
    type: "appointment",
  },
  {
    date: currentDate,
    status: "Received",
    count: 105,
    source: "Cars.com",
    attampt: "3rd Attampt",
    type: "appointment",
  },
]

export const appointmentsAnalytics = (data, range) => {
  // actual
  const _data = (data || []).map(el => ({
    ...el,
    previous: el.prevCount,
    change: getChange(el.count, el.prevCount),
  }))

  return _data
}

export const engagedConverted = (data, range) => {
  const engaged = 1000
  const converted = 800

  if (engaged === 0 && converted === 0) return null
  return [
    { type: "Engaged", value: engaged },
    { type: "Converted", value: converted },
  ]
}

const EngagementDashboardView = ({ location }) => {
  // const { data: respSummary, loading } = useQuery(GET_CAMPAIGN_LEAD_SUMMARY, {
  //   fetchPolicy: "network-only",
  //   variables: {
  //     range: 180,
  //   },
  // })
  // const summaryData = resp?.getCampaignLeadSummary?.leadMessageCount || []
  const { data: resp, loading } = useQuery(GET_ENGAGEMENT_ANALYTICS, {
    fetchPolicy: "network-only",
    variables: {
      range: 180,
    },
  })
  const analyticsData = resp?.getEngagementAnalytics || []

  const { data: result } = useQuery(GET_ENGAGEMENT_LEADS_ANALYTICS, {
    fetchPolicy: "network-only",
    variables: {
      range: 180,
    },
  })
  const analyticsLeadsData = result?.getEngagementLeadAnalytics || []

  const { data: resLeads, loading: leadLoading } = useQuery(
    GET_LEADS_ANALYTICS,
    {
      fetchPolicy: "network-only",
      variables: {
        range: 180,
      },
    }
  )
  const leadsAnalyticsData = resLeads?.getLeadAnalytics || []

  const { data: resAppointments, loading: aptLoading } = useQuery(
    GET_APPOINTMENT_ANALYSIS,
    {
      fetchPolicy: "network-only",
      variables: {
        range: 180,
      },
    }
  )

  const { data: resAppSource, loading: aptSourceLoading } = useQuery(
    GET_APPOINTMENTS_BY_SOURCE,
    {
      fetchPolicy: "network-only",
      variables: {
        range: 180,
      },
    }
  )

  const { data: resAppSalesPerson, loading: aptSalesPersonLoading } = useQuery(
    GET_APPOINTMENTS_BY_SALES_PERSON,
    {
      fetchPolicy: "network-only",
      variables: {
        range: 180,
      },
    }
  )

  const conductedLeads = filterByStatus(analyticsLeadsData)([
    "SENT",
    "OPT_PUT",
    "DELIVERED",
    "RESPONDED",
    "ENGAGED",
  ])

  const receivedRespLeads = filterByStatus(analyticsLeadsData)(["RESPONDED"])
  let leadsData = [...analyticsLeadsData]
  leadsAnalyticsData.forEach(el => {
    leadsData.push({
      ...el,
      status: "N/A",
      date: "",
    })
  })

  return (
    // <PaddedCol>
    <>
      
      <Row>
        <OptionsWrapper
          changes
          lambda={activityAnalysis}
          data={analyticsData}
          loading={loading}
          title="Engagement Analysis"
        >
          {({ data }) => <ProspectAnalytics data={data} />}
        </OptionsWrapper>
      </Row>
      <ContainerGroup>
        <OptionsWrapper
          span={12}
          periods
          lambda={responseByEngagement}
          loading={loading}
          data={analyticsData}
          graphTypes={[BAR]}
          title="Responses Received By Attempt"
        >
          {({ data, period, graphType }) => (
            <>
              {graphType === BAR && (
                <BarChart
                  period={period}
                  id="resp-received-by-source"
                  //horizontal={true}
                  data={data}
                />
              )}
              {graphType === LINE && <LineChart data={data} />}
            </>
          )}
        </OptionsWrapper>
        <OptionsWrapper
          span={12}
          periods
          lambda={leadBySource}
          loading={loading}
          data={receivedRespLeads}
          graphTypes={[BAR, LINE]}
          title="Responses By Source"
        >
          {({ data, period, graphType }) => (
            <>
              {graphType === BAR && (
                <BarChart
                  period={period}
                  id="response-source"
                  horizontal={true}
                  data={data}
                />
              )}
              {graphType === LINE && <LineChart data={data} />}
            </>
          )}
        </OptionsWrapper>
      </ContainerGroup>
      <Row>
        <Col>
          <OptionsWrapper
            changes
            lambda={appointmentsAnalytics}
            data={resAppointments?.getAppointmentAnalysis || []}
            loading={aptLoading}
            title="Appointments Analysis"
            //span={12}
          >
            {({ data }) => <ProspectAnalytics data={data} />}
          </OptionsWrapper>
        </Col>
      </Row>
      <ContainerGroup>
        <OptionsWrapper
          span={12}
          periods
          lambda={leadBySource}
          loading={aptSourceLoading}
          data={resAppSource?.getAppointmentSource || []}
          graphTypes={[BAR, LINE]}
          title="Appointments By Source "
        >
          {({ data, period, graphType }) => (
            <>
              {graphType === BAR && (
                <BarChart
                  period={period}
                  id="appointment-source"
                  horizontal={false}
                  data={data}
                />
              )}
              {graphType === LINE && <LineChart data={data} />}
            </>
          )}
        </OptionsWrapper>
        <OptionsWrapper
          span={12}
          periods
          lambda={appointmentsBySalesPerson}
          loading={false}
          data={resAppSalesPerson?.getAppointmentSalesperson || []}
          graphTypes={[BAR, LINE]}
          title="Appointments By Sales Person"
        >
          {({ data, period, graphType }) => (
            <>
              {graphType === BAR && (
                <BarChart
                  period={period}
                  id="appointment-by-sales-person"
                  horizontal={true}
                  data={data}
                />
              )}
              {graphType === LINE && <LineChart data={data} />}
            </>
          )}
        </OptionsWrapper>
      </ContainerGroup>
      <Row>
        <Col>
          <OptionsWrapper
            changes
            lambda={leadAnalysis}
            data={leadsData}
            loading={leadLoading}
            title="Lead Analysis"
            //span={12}
          >
            {({ data }) => <ProspectAnalytics data={data} width={"70%"} />}
          </OptionsWrapper>
        </Col>
      </Row>
      <ContainerGroup>
        <OptionsWrapper
          span={12}
          periods
          lambda={leadBySource}
          loading={leadLoading}
          data={leadsAnalyticsData}
          graphTypes={[BAR, LINE]}
          title="Lead By Source "
        >
          {({ data, period, graphType }) => (
            <>
              {graphType === BAR && (
                <BarChart
                  period={period}
                  id="lead-by-source"
                  horizontal={false}
                  data={data}
                />
              )}
              {graphType === LINE && <LineChart data={data} />}
            </>
          )}
        </OptionsWrapper>
        <OptionsWrapper
          span={12}
          periods
          lambda={leadBySource}
          loading={leadLoading}
          data={conductedLeads}
          graphTypes={[BAR, LINE]}
          title="Lead Contacted By Source "
        >
          {({ data, period, graphType }) => (
            <>
              {graphType === BAR && (
                <BarChart
                  period={period}
                  id="lead-conducted-by-source"
                  horizontal={false}
                  data={data}
                />
              )}
              {graphType === LINE && <LineChart data={data} />}
            </>
          )}
        </OptionsWrapper>
      </ContainerGroup>
    </>
    // </PaddedCol>
  )
}

export default EngagementDashboardView
