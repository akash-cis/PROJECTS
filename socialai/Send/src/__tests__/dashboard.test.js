import { toTreeMapCompatible } from "../components/analytics/dashboard"

const dating = [
  {
    make: "Tesla",
    model: "Model 8",
    count: 5,
  },
  {
    make: "Ford",
    model: "Model 8",
    count: 10,
  },
  {
    make: "Tesla",
    model: "Model 9",
    count: 9,
  },
  {
    make: "Ford",
    model: "Model 9",
    count: 4,
  },
  {
    make: "Tesla",
    model: "Model 5",
    count: 5,
  },
]

test("processes data correctly", () => {
  expect(toTreeMapCompatible(dating)).toMatchSnapshot()
})
