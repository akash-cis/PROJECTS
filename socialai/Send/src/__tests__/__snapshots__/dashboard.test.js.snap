// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`processes data correctly 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "count": 10,
        "name": "Model 8",
      },
      Object {
        "count": 4,
        "name": "Model 9",
      },
    ],
    "model": "Model 8",
    "name": "Ford",
    "total": 14,
  },
  Object {
    "children": Array [
      Object {
        "count": 5,
        "name": "Model 8",
      },
      Object {
        "count": 9,
        "name": "Model 9",
      },
      Object {
        "count": 5,
        "name": "Model 5",
      },
    ],
    "model": "Model 8",
    "name": "Tesla",
    "total": 19,
  },
]
`;
